src/asynchronous/libasync/events.d

```diff
@@ -238,9 +239,19 @@ package class LibasyncEventLoop : EventLoop
             auto socket1 = new Socket(cast(socket_t) connection.socket,
                 socket.addressFamily);
 
-            auto transport = new LibasyncTransport(this, socket1, connection);
+            LibasyncTransport transport;
+			if (sslContext is null)
+			{
+				transport =  new LibasyncTransport(this, socket1, connection);
+				transport.setProtocol(protocolFactory());
+			}
+			else
+			{
+				auto sslProtocol = createSslProtocol(this, protocolFactory(), sslContext);
+				transport =  new LibasyncTransport(this, socket1, connection);
```

Comment:
I currently wanted and tried to use ```makeTLSTransport`` and ```makeSocketTransport``` but ```makeSocketTransport``` isn't written in the same way like in Python, so I wasn't sure if it does the same thing. For me something like Python original code would look better:
```D
if (sslContext is null)
 transport = makeSocketTransport(...)
else
 transport = makeTLSTransport(...)
```
Will it work this way?